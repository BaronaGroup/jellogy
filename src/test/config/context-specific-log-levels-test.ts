import {prepareTestKarhu} from '../test-utils'

describe('context-specific-log-levels-test', () => {
  const karhuTest = prepareTestKarhu({
    contextSpecificLogLevels: new Map<string | RegExp, string>([
      ['very-important', 'DEBUG'],
      ['long-match', 'DEBUG'],
      [/match/, 'INFO']
    ]),
    defaultLogLevel: 'WARN'
  })

  it('allows overriding the default log level for individual contexts', karhuTest((karhu, output) => {
    karhu.context('not-so-important').debug('This should not show up')
    karhu.context('very-important').debug('This should show up')

    expect(output.tracked).toMatchSnapshot()
  }))

  it('regexp are supported', karhuTest((karhu, output) => {
    karhu.context('this-should-match').info('This should show up')
    karhu.context('this-should-match').debug('This should not show up')
    expect(output.tracked).toMatchSnapshot()
  }))

  it('perfect match is preferred over regexp', karhuTest((karhu, output) => {
    karhu.context('long-match').debug('This should show up')
    expect(output.tracked).toMatchSnapshot()
  }))
})
