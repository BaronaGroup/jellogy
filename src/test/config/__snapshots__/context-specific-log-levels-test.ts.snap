// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`context-specific-log-levels-test allows overriding the default log level for individual contexts 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [
    Array [
      "[1] [95mDEBUG very-important This should show up[39m",
      "DEBUG",
      "very-important",
    ],
  ],
  "INFO" => Array [],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;
